.github/workflows/matrix.yml

```diff
@@ -0,0 +1,112 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v4.3.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2
+
+      - name: Install additional Helm repos
+        run: |
+          helm repo add bitnami https://charts.bitnami.com/bitnami
+          helm repo add digicatapult https://digicatapult.github.io/helm-charts
+
+      # only keep changed logic get rid of named_charts logic
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed --target-branch ${{ github.event.repository.default_branch || 'main' }})
+          if [[ -n "$changed" ]]; then
+            echo "changed=true" >> $GITHUB_OUTPUT
+          else
+            echo "changed=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Run chart-testing (lint)
+        run: |
+          ct lint \
+            --target-branch ${{ github.event.repository.default_branch }}
+        if: steps.list-changed.outputs.changed == 'true'
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1
+        if: steps.list-changed.outputs.changed == 'true'
+
+      - name: Install dependencies postgress and sqnc-node manually
+        if: steps.list-changed.outputs.changed == 'true'
+        run: |
+          helm install postgres bitnami/postgresql \
+            --version 15.1.0 \
+            --set auth.username="identity_service" \
+            --set auth.password="test-password" \
+            --set auth.postgresPassword=test-password \
+            --set auth.database="user" \
+            --set architecture="standalone" \
+            --wait
+
+            kubectl wait --for=condition=ready pod -l app.kubernetes.io/name=postgresql --timeout=300s
+
+          POSTGRES_PASSWORD=$(kubectl get secret postgres-postgresql -o jsonpath="{.data.postgres-password}" | base64 -d)
+          kubectl run postgres-postgresql-client --rm --tty -i --restart='Never' \
+          --namespace default \
+          --image docker.io/bitnami/postgresql:15.1.0 \
+          --env="PGPASSWORD=$POSTGRES_PASSWORD" \
+          --command -- psql -h postgres-postgresql -U postgres -d user -c '\l'
+
+
+          helm install sqnc-node ./charts/sqnc-node \
+            --set fullnameOverride=sqnc-node \
+            --set service.port=9944 \
+            --set persistence.enabled=true \
+            --wait
+
+            kubectl wait --for=condition=ready pod -l app.kubernetes.io/name=sqnc-node --timeout=300s
+
+          echo "Postgres pod status:"
+          kubectl get pods -l app.kubernetes.io/name=postgresql
+
+          echo "Postgres logs:"
+          kubectl logs -l app.kubernetes.io/name=postgresql || true
```

Comment:
Split these two dependency installs into two separate steps and remove all the unnecessary debug here.